docs/usage/WritingCustomMiddleware.md

```diff
@@ -0,0 +1,146 @@
+---
+id: writing-custom-middleware
+title: Writing Custom Middleware
+---
+
+# Writing Custom Middleware
+
+:::tip What You'll Learn
+
+- When to use custom middleware
+- Standard patterns for middleware
+- How to make sure that your middleware is compatible with other Redux projects
+
+:::
+
+Middleware in Redux can be mainly used to either
+
+- create side effects for actions, or to
+- modify or cancel actions.
+
+Most use cases fall into the first category: For example [Redux-Saga](https://github.com/redux-saga/redux-saga/), [redux-observable](https://github.com/redux-observable/redux-observable), and [RTK listener middleware](https://redux-toolkit.js.org/api/createListenerMiddleware) all create side effects that react to actions. These examples also show that this is a very common need: To be able to react to an action other than with a state change.
+
+The most obvious example for modifying actions is [Redux Thunk](https://github.com/reduxjs/redux-thunk), which transforms a function returning an action into an action by calling it. But it can also be used to e.g. enhance an action with information from the state or from an external input.
```

Comment:
:hand: I'd add a third category - "teaching `dispatch` how to understand values that are _not_ plain action objects", and put thunks into that.  "Modifying an action" to me would be something like debouncing / throttling / gating an action, or literally mutating it.